quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java

```diff
@@ -177,6 +187,48 @@ private void validateTlsAvailable(GlobalConfiguration config) {
 
     }
 
+    private void configureRemoteStores(ConfigurationBuilderHolder builder) {
+        if (builder.getNamedConfigurationBuilders().get(USER_SESSION_CACHE_NAME).clustering().cacheMode().isDistributed()) {
+            //if one of remote store command line parameters is defined, all other are required, otherwise assume it'd configured via xml only
+            if (Configuration.getOptionalKcValue(CACHE_REMOTE_HOST_PROPERTY).isPresent() ||
+                Configuration.getOptionalKcValue(CACHE_REMOTE_PORT_PROPERTY).isPresent() ||
+                Configuration.getOptionalKcValue(CACHE_REMOTE_USERNAME_PROPERTY).isPresent() ||
+                Configuration.getOptionalKcValue(CACHE_REMOTE_PASSWORD_PROPERTY).isPresent()) {
+
+                String cacheRemoteHost = requiredStringProperty(CACHE_REMOTE_HOST_PROPERTY);
+                String cacheRemotePort = requiredStringProperty(CACHE_REMOTE_PORT_PROPERTY);
+                String cacheRemoteUsername = requiredStringProperty(CACHE_REMOTE_USERNAME_PROPERTY);
+                String cacheRemotePassword = requiredStringProperty(CACHE_REMOTE_PASSWORD_PROPERTY);
+
+                DISTRIBUTED_REPLICATED_CACHE_NAMES.forEach(cacheName -> {
+                    PersistenceConfigurationBuilder persistenceCB = builder.getNamedConfigurationBuilders().get(cacheName).persistence();
+
+                    //if specified via command line -> cannot be defined in the xml file
+                    if (!persistenceCB.stores().isEmpty()) {
+                        throw new RuntimeException(String.format("Remote store for '%s' is already configured via CLI parameters. It should not be present in XML file a.", cacheName));
+                    }
+
+                    persistenceCB.addStore(RemoteStoreConfigurationBuilder.class)
+                            .rawValues(true)
+                            .shared(true)
+                            .remoteCacheName(cacheName)
+                            .connectionPool()
+                                .maxActive(16)
+                                .exhaustedAction(ExhaustedAction.CREATE_NEW)
+                            .remoteSecurity()
+                                .authentication()
+                                    .serverName("infinispan")
+                                    .username(cacheRemoteUsername)
+                                    .password(cacheRemotePassword)
+                                    .realm("default")
```

Comment:
@pruivo - what is the default that Infinispan uses? 

From a security point of view, I'd like to see `SCRAM-SHA-256` or better.

Reply:
https://github.com/infinispan/infinispan/blob/b85bf4ee74175cf70b04df8a0a64aec9ee56c363/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/AuthenticationConfigurationBuilder.java#L35

It looks like `SCRAM-SHA-512`

Reply:
I hope it's fine to put the default from Infinispan here: 

```java
.saslMechanism(AuthenticationConfigurationBuilder.DEFAULT_MECHANISM)
```

Please let me know if the hardcoded string would be better.

Reply:
Use this constant instead: `org.wildfly.security.sasl.util.SaslMechanismInformation.Names.SCRAM_SHA_512`

The reason is that `SCRAM-*` requires `wildfly-elytron-sasl-*` dependencies and if they are removed by mistake, the compilation would fail in this line, signaling a problem right away. Otherwise, it would fail at runtime.

Let me know if that makes sense :)